platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/HttpClientBuilder.java

```diff
@@ -0,0 +1,200 @@
+  }
+
+  public boolean useTls() {
+    return StringUtils.startsWithIgnoreCase(
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>) () -> System.getProperty(SOLR_HTTP_URL)),
+        "https");
+  }
+
+  private CredentialsProvider getCredentialsProvider() {
+    CredentialsProvider provider = new BasicCredentialsProvider();
+
+    org.apache.http.auth.UsernamePasswordCredentials credentials =
+        new org.apache.http.auth.UsernamePasswordCredentials(
+            AccessController.doPrivileged(
+                (PrivilegedAction<String>) () -> System.getProperty("solr.username")),
+            getPlainTextSolrPassword());
+    provider.setCredentials(AuthScope.ANY, credentials);
+    return provider;
+  }
+
+  private static SSLContext getSslContext() {
+    final Boolean check =
+        AccessController.doPrivileged(
+            (PrivilegedAction<Boolean>)
+                () ->
+                    (System.getProperty(KEY_STORE) == null
+                        || System.getProperty(KEY_STORE_PASS) == null
+                        || System.getProperty(TRUST_STORE) == null
+                        || System.getProperty(TRUST_STORE_PASS) == null));
+
+    if (check) {
+      throw new IllegalArgumentException("KeyStore and TrustStore system properties must be set.");
+    }
+
+    final KeyStore[] trustStore = new KeyStore[1];
+    final KeyStore[] keyStore = new KeyStore[1];
+
+    AccessController.doPrivileged(
+        (PrivilegedAction<Object>)
+            () -> {
+              trustStore[0] =
+                  getKeyStore(
+                      System.getProperty(TRUST_STORE), System.getProperty(TRUST_STORE_PASS));
+              keyStore[0] =
+                  getKeyStore(System.getProperty(KEY_STORE), System.getProperty(KEY_STORE_PASS));
+              return null;
+            });
+
+    // IDE thinks this variable does not need to be initialized. However, because of the try/catch
+    // block, there is a situation where it could end up uninitialized.
+    SSLContext sslContext = null;
+
+    try {
+      sslContext =
+          SSLContexts.custom()
+              .loadKeyMaterial(
+                  keyStore[0],
+                  AccessController.doPrivileged(
+                          (PrivilegedAction<String>) () -> System.getProperty(KEY_STORE_PASS))
+                      .toCharArray())
+              .loadTrustMaterial(trustStore[0])
+              .useTLS()
+              .build();
+    } catch (UnrecoverableKeyException
+        | NoSuchAlgorithmException
+        | KeyStoreException
+        | KeyManagementException e) {
+      throw new IllegalArgumentException(
+          "Unable to use javax.net.ssl.keyStorePassword to load key material to create SSL context for Solr client.");
+    }
+
+    sslContext.getDefaultSSLParameters().setNeedClientAuth(true);
+    sslContext.getDefaultSSLParameters().setWantClientAuth(true);
```

Comment:
This 2 lines probably ought to be in the `try` block as technically `sslContext` could be null outside of it.